pkg/registry/core/pod/storage/eviction.go

```diff
@@ -130,13 +130,55 @@ func (r *EvictionREST) Create(ctx context.Context, name string, obj runtime.Obje
 
 	// Evicting a terminal pod should result in direct deletion of pod as it already caused disruption by the time we are evicting.
 	// There is no need to check for pdb.
-	if pod.Status.Phase == api.PodSucceeded || pod.Status.Phase == api.PodFailed {
-		_, _, err = r.store.Delete(ctx, eviction.Name, rest.ValidateAllObjectFunc, deletionOptions)
-		if err != nil {
-			return nil, err
+	if canIgnorePDB(pod) {
+		deleteRefresh := false
+		continueToPDBs := false
+		// Preserve current deletionOptions if we need to fall through to checking PDBs later.
+		preservedDeletionOptions := deletionOptions.DeepCopy()
```

Comment:
minor: rename `deleteOptions` to `originalDeleteOptions` and do something like

```go
retry.RetryOnConflict{
  // avoid mutating original delete options.
  currDeleteOptions := originalDeleteOptions.DeepCopy()
}
```

The reassignment of bits down below is making me cross-eyed.